### PR TITLE
abs() namespace ambiguity fix for Windows

### PR DESCRIPTION
**Pull request by kimberlyroche**
Tested on Win10/R 3.6.0 and MacOS X 10.14/R 3.6.0

### CODE REVIEW

**Comment by jsilve24** [comment]
Thank you!

**Comment by jsilve24** [comment]
For future reference: Here is Kim's Write up and multiple solutions that could have fixed this problem:


Tl;dr I think the Windows failed Cholesky's are solved. It's a crazy one. The bug seemed to be the behavior of abs() in calcLogLik and different namespace resolution on Windows/Mac. Apparently I'm not the first person to find this; the Rcpp folks are aware and seem to consider it sort of caveat emptor: https://github.com/RcppCore/Rcpp/issues/167

On both operating systems, assuming Rcpp >= 0.8.3, the call to abs() hits Rcpp::abs(), which expects a NumericVector argument, gets a scalar instead, and passes the buck to the C/C++ libraries.

On MacOS, which is using clang/clang++ via Xcode as the default compiler (not gcc/g++), some chain of fortuitous configuration settings seems to cause abs() to hit the C++11 stdlib version, which is happy to take a float and return a float. Life is good.

On Windows, as ever, a batshit thing happens instead. Whether you compile with C++11 or not, unless you explicitly call the stdlib version of abs (via std::abs()), the compiler falls back to the ancient C abs() which returns an int, so all floats are truncated (!), e.g. abs(1.3) == abs(-1.3) == 1 and so on. For the most part the optimization survives this but is always off, as you can imagine. Normally, the integer version of abs() would throw an error when passed a float or double as an argument but for some reason, when concealed under Rcpp, it does not, so it's probably still a cause for official complaint.

Anyway, possible fixes:
(1) An explicit include of <cmath> in PibbleCollapsed.h or its included headers in order to overload C's abs() with C++'s std::abs().
(2) A change in Makevars.win to force CXX_STD = CXX11 on by default and a change from abs() to std::abs() in PibbleCollapsed.h.
(3) Just change abs() to fabs() (the base C function for floats/doubles) and be done with it.

The last one is arguably the most ass-covering and has the benefit of being the single smallest bug fix imaginable: a one-letter update. Maddening.

I'll test those changes tomorrow on Win/Mac, confirm everything is good, do a dev pull request, and then hopefully no special instructions at all needed for Windows.
